internal/expr/parser.go

```diff
@@ -502,35 +577,100 @@ func (p *Parser) parseColumn() (columnAccessor, bool, error) {
 	return columnAccessor{}, false, nil
 }
 
-func (p *Parser) parseTargetType() (valueAccessor, bool, error) {
+func (p *Parser) parseTargetType() (keyedAccesor, bool, error) {
+	startLine := p.lineNum
+	startCol := p.colNum()
+
 	if p.skipByte('&') {
+		// Using a slice as an output is an error, we add the case here to
+		// improve the error message.
+		if st, ok, err := p.parseSliceAccessor(); err != nil {
+			return keyedAccesor{}, false, err
+		} else if ok {
+			return keyedAccesor{}, false, errorAt(fmt.Errorf("cannot use slice syntax %q in output expression", st), startLine, startCol, p.input)
+		}
 		return p.parseTypeName()
 	}
 
-	return valueAccessor{}, false, nil
+	return keyedAccesor{}, false, nil
+}
+
+// parseUNumber parses a non-negative number composed of one or more digits.
+func (p *Parser) parseUNumber() (*uint64, bool) {
+	mark := p.pos
+	if !p.skipNumber() {
+		return nil, false
+	}
+	n, _ := strconv.ParseUint(p.input[mark:p.pos], 10, 64)
+	return &n, true
+}
+
+// parseSliceAccessor parses a slice range composed of two indexes of the form
```

Comment:
It might be worth pointing out in the comment that it can return a `sliceIndex` or a `sliceRange`, since the `valueAccessor` is an interface.